source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs

```diff
@@ -9,84 +10,160 @@
 
 namespace GamesSizeCalculator.SteamSizeCalculation
 {
-    public class SteamSizeCalculator
+    public class SteamSizeCalculator : IOnlineSizeCalculator
     {
         private ILogger logger = LogManager.GetLogger();
         public ISteamApiClient SteamApiClient { get; }
+        public ISteamAppIdUtility SteamAppIdUtility { get; }
+        public bool IncludeDLC { get; set; }
+        public bool IncludeOptional { get; set; }
+        public bool HandleNonSteamGames { get; set; }
+        public string[] RegionalWords { get; set; }
+        public string[] RegionalWordsBlacklist { get; set; }
+        public string ServiceName { get; } = "Steam";
 
-        public SteamSizeCalculator(ISteamApiClient steamApiClient)
+        public SteamSizeCalculator(ISteamApiClient steamApiClient, ISteamAppIdUtility steamAppIdUtility, bool includeDLC, bool includeOptional, bool handleNonSteamGames, string[] regionalWords, string[] regionalWordsBlacklist)
         {
             SteamApiClient = steamApiClient;
+            SteamAppIdUtility = steamAppIdUtility;
+            IncludeDLC = includeDLC;
+            IncludeOptional = includeOptional;
+            HandleNonSteamGames = handleNonSteamGames;
+            RegionalWords = regionalWords;
+            RegionalWordsBlacklist = regionalWordsBlacklist;
```

Comment:
I'm wondering if instead of setting individual fields from settings, it would be possible to set a settings field with the plugin settings and access the properties directly